foundationdb-kubernetes-sidecar/sidecar.py

```diff
@@ -305,6 +326,24 @@ def is_at_least(self, target_version):
             and self.minor_version[1] >= target_version[1]
         )
 
+    @classmethod
+    def extract_desired_ip(cls, pattern, string):
+        if string == "":
+            return string
+
+        ips = string.split(",")
+        matching_ips = [ip for ip in ips if pattern.search(ip)]
+        if len(matching_ips) == 0:
+            raise Exception("Failed to find IP matching %s in %s" % (pattern, ips))
+        ip = matching_ips[0]
+        if ":" in ip:
+            ip = "[%s]" % ip
+        return ip
```

Comment:
The returned IP address for IPv6 will be invalid since `[::]` is not an valid IPv6 address

Reply:
We're enclosing the addresses in brackets because everywhere we are using it in the monitor conf, it gets paired with a port, so it needs the brackets. We could modify the monitor conf to contain the brackets, but I think that would lead to more challenges with the rollouts.

Reply:
I think for the current state but we might to revisit this. Sadly Python doesn't have https://golang.org/pkg/net/#JoinHostPort (at least to my knowledge) and I remember doing the same fun stuff: https://github.com/FoundationDB/fdb-kubernetes-operator/pull/341/files#diff-361ca47879a27b8387987496c32b845dbea647c753333511aebc6d93a19be6edR505-R521 I think I just moved it to a later state to prevent parsing the address for the web server again.